pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/ComparableOneOf.java

```diff
@@ -44,8 +45,8 @@ public boolean equals(Object o) {
 
     @Override
     public int hashCode() {
-        // name().hashCode() and NOT Enum.hashCode() because the latter changes between runs
-        return (31 + kind.name().hashCode()) * 31 + (value == null ? 0 : value.hashCode());
+        return (31 + Integer.hashCode(((EnumWithProtoMetadata)kind).protoOrdinal())) * 31
```

Comment:
Nit: I don't think adding 31 provides a lot of value. It basically just adds a constant to the result. But it does not hurt.

Reply:
The `31` comes from an implementation of `Objects.hashCode([])` which we removed nad replaced with this code because it created a new array and hurt the performance. However, we wanted to keep the hash codes values intact, hence the `31`.